### ISSUE TITLE
Pushing new story 

### ISSUE THREAD

**Issue by sandinmyjoints** [org-pivotal/org-pivotal#7]
I'm authenticated b/c I can successfully fetch stories, but when I type out a new one and try to push it, I get:

```
[error] request--callback: peculiar error: 404
[error] request-default-error-callback: https://www.pivotaltracker.com/services/v5/projects/1967621/stories/ error
```

I did this both with no ID (aiming to create a new story), and also by creating an empty ticket using the Pivotal interface, grabbing the ID from that, and putting into my org file, and then pushing. Got the same error both ways.

**Comment by qhuyduong** [comment]
@sandinmyjoints thanks for reporting. I will talk a look

**Comment by sandinmyjoints** [comment]
It's a nice library, btw! org-mode seems like a natural fit for working with Pivotal in emacs. I'd like to incorporate this into my workflow, and if I get some extra time, I will try to get more familiar with the code and open a PR.

**Comment by adityaathalye** [comment]
Attempting to fetch tasks results in an error similar to what OP observes when they try to push a story.

I'm using Emacs 26.1, with org-pivotal `20200607.1505`, on Ubuntu 18.04 LTS.

TL;DR: `org-get-entry` called inside `org-pivotal-pull-story-tasks` is unable to locate the `ID` property for a story, causing a broken URL to be generated.

The API call itself works if the url is constructed. Verification below (project/task IDs altered for privacy):

On calling `C-( t`, message buffer shows this error:
```text
[error] request--callback: peculiar error: 400
[error] request-default-error-callback: https://www.pivotaltracker.com/services/v5/projects/1234567/stories//tasks error
```

The slug between stories//tasks is empty, whereas it should be the task ID, based on [the story-tasks API](https://www.pivotaltracker.com/help/api/rest/v5#Story_Tasks).

The org-pivotal code looked fine to me, so I traced a bit, and narrowed down to `org-entry-get`. I observed that org-entry-get was not able to look up the "ID" property:

```text
1 -> (org-pivotal-pull-story-tasks)
| 2 -> (org-entry-get 364 "project-id" t)
| 2 <- org-entry-get: "1234567"
| 2 -> (org-entry-get 364 "ID" t)
| 2 <- org-entry-get: nil
1 <- org-pivotal-pull-story-tasks: nil
```

Next, 
* I set inherit to `t` in `org-pivotal-pull-story-tasks`
```elisp
(defun org-pivotal-pull-story-tasks ()
  "Pull current story's tasks."
  (interactive)
  (org-pivotal--append-tasks-to-current-story
   (org-pivotal-api--fetch-story-tasks
    (org-entry-get (point) "project-id" t)
    (org-entry-get (point) "ID" t))))
```
* And copied the story `ID` into the global Properties drawer (as `#+PROPERTY: ID 987654321`).

Now, calling `C-( t` pulls in the tasks if I do this:
 - Expand the `:PROPERTIES:` drawer of the story (this causes the now-global `ID` to be inherited)
 - Move point after `:END:` tag of the property drawer
 - `C-( t`
 - Tasks are fetched and inserted at point.

```text
======================================================================
1 -> (org-entry-get nil "LOG_INTO_DRAWER" inherit t)
1 <- org-entry-get: nil
======================================================================
1 -> (org-pivotal-pull-story-tasks)
| 2 -> (org-entry-get 643 "project-id" t)
| 2 <- org-entry-get: "1234567"
| 2 -> (org-entry-get 643 "ID" t)
| 2 <- org-entry-get: "987654321"
1 <- org-pivotal-pull-story-tasks: ((nil) (nil) (nil) (nil) (nil) (nil))
```

**Comment by adityaathalye** [comment]
@qhuyduong I realise why [the problem I described](https://github.com/org-pivotal/org-pivotal/issues/7#issuecomment-647474666) is happening. 

It appears that every property needs to be in a single line. Also, all properties need to be contiguous, within a property drawer. If _any_ of the property items has a value spanning multiple lines, `org-entry-get` fails.

I found out by evaluating `org-buffer-property-keys` in my file, with the two examples below: 
```
* Can org fetch properties?
  #+BEGIN_SRC elisp
  (org-buffer-property-keys)
  #+END_SRC

  #+RESULTS:
  <property drawer items will appear here IFF `org-buffer-property-keys` succeeds.>
```

**The following format works:**
```
* Started A really nice title to the story
  :PROPERTIES:
  :ID:       123456789
  :Type:     Feature
  :Points:   2
  :Updated:  2020-01-02T04:13:01Z
  :URL:      https://www.pivotaltracker.com/story/show/123456789
  :Description: A description in pivotal that fits into a single line (no line breaks or newlines).
  :Labels:
  :END:
  - [x] A task that got done
  - [x] Another task that got done
  - [ ] A task that remains to be done
```

**The following format fails.** Org will complain saying "Malformed drawer at 123, repair? (y or n)":
```
* Started A really nice title to the story
  :PROPERTIES:
  :ID: 123456789
  :Type: Feature
  :Points: 2
  :Updated: 2020-01-02T04:13:01Z
  :URL: https://www.pivotaltracker.com/story/show/123456789
  :Description: A description in pivotal with multiple lines:
1. Bullet point
2. Another bullet point
3. Yet another bullet point
  :Labels:
  :END:
```

**Comment by qhuyduong** [comment]
@adityaathalye thank you. I noticed this issue before but not having time to fix it yet. PR is welcome!

**Comment by adityaathalye** [comment]
@qhuyduong I posted the discovery so I don't forget, and so it's available to anyone else who might pick it first.

I was trying some things today. I'll send PR if I can figure out a good fix!

**Comment by adityaathalye** [comment]
P.S. Thank you for making the project :)

**Comment by sandinmyjoints** [comment]
It is possible to get org to read multiple lines for a property like this:

```
* Started A really nice title to the story
  :PROPERTIES:
  :ID: 123456789
  :Type: Feature
  :Points: 2
  :Updated: 2020-01-02T04:13:01Z
  :URL: https://www.pivotaltracker.com/story/show/123456789
  :Description: A description in pivotal with multiple lines:
  :Description+: 1. Bullet point
  :Description+: 2. Another bullet point
  :Description+: 3. Yet another bullet point
  :Labels:
  :END:
```

However, it does not preserve the newlines but concats all the lines, which isn't practical for ticket descriptions since it only allows one paragraph of text and ruins any formatting.

Since multiple lines seems not at all to work nicely with org properties, would it be possible to treat the description field differently, and instead look for it as a second-level headline and whatever contents belong to it? Something like:

```
* Started A really nice title to the story
  :PROPERTIES:
  :ID: 123456789
  :Type: Feature
  :Points: 2
  :Updated: 2020-01-02T04:13:01Z
  :URL: https://www.pivotaltracker.com/story/show/123456789
  :Labels:
  :END:
* Description
A description in pivotal with multiple lines:

1. Bullet point
2. Another bullet point
3. Yet another bullet point
```

In my experience, descriptions can contain dozens of lines of text, and markdown formatting including code blocks, which seems more natural in this format than in a property.

**Comment by sandinmyjoints** [comment]
Fixed by https://github.com/org-pivotal/org-pivotal/pull/10 (original issue)
Multi-line description probably deserves its own issue.
